lib/cloudregister/registerutils.py

```diff
@@ -717,6 +768,73 @@ def same_registry_auth_content(content, smt):
         return one_auths_key and auth_token in content.get('auths').values()
 
 
+# ----------------------------------------------------------------------------
+def unset_env_vars():
+    """Remove the registry environment variables."""
+    env_vars = ['REGISTRY_AUTH_FILE', 'DOCKER_CONFIG']
+    if not os.path.exists(BASHRC_LOCAL_PATH):
+        logging.info('%s file does not exist' % BASHRC_LOCAL_PATH)
+        # remove the enviroment variables from the env, if present
+        return True
+
+    lines, modified, preserved_failed, mv_backup = clean_bashrc_local(env_vars)
+    succeeded = True
+    if modified:
+        # we could access the file and registry env vars were found and removed
+        succeeded = update_bashrc(''.join(lines), 'w')
+    else:
+        # Either we could access the file and no registry env vars were found
+        # or we could not access the file and attempted to create a backup
+        # if backup scenario, log is already populated
+        if not mv_backup:
+            # we could access the bashrc local file and
+            # no env vars were found
+            logging.info(
+                'Environment variables not present in %s' % BASHRC_LOCAL_PATH
+            )
+            succeeded = True
+        elif preserved_failed:
+            # we could not access the bashrc local file
+            # the attempt to create a backup failed
+            succeeded = False
+
+    return succeeded
+
+
+# ----------------------------------------------------------------------------
+def clean_bashrc_local(env_vars):
+    """
+    Clean the registry env vars, if any, from the BASHRC_LOCAL_PATH file
+
+    :returns:
+        - bashrc_local_new_lines: list - the new lines after cleaning
+        - modified: bool - whether the content of the file was modified
+        - preserved_failed: bool - if accessing the file failed,
+                                   whether preserve op failed
+        - mv_backup: bool: whether the backup file operation suceeded
+    """
+    bashrc_local_lines = []
+    try:
+        with open(BASHRC_LOCAL_PATH, 'r') as bashrc_local:
+            bashrc_local_lines = bashrc_local.readlines()
+    except OSError as error:
+        logging.info('Could not open %s: %s' % (BASHRC_LOCAL_PATH, error))
+        failed = __mv_file_backup(BASHRC_LOCAL_PATH)
+        return [], False, failed, True
+
+    bashrc_local_new_lines = []
+    for bashrc_line in bashrc_local_lines:
+        skip = False
+        for env_var in env_vars:
+            if not bashrc_line.startswith('#') and env_var in bashrc_line:
+                skip = True
+        if not skip:
+            bashrc_local_new_lines.append(bashrc_line)
+
+    modified = bashrc_local_lines != bashrc_local_new_lines
```

Comment:
This potentially long and therefore expensive string comparison can be skipped, we already know when we modified the, which happens when we do not re-add the entries for the environment variables.